src/functions/indGraphSkinny.jl

```diff
@@ -0,0 +1,54 @@
+# This implements the variant when A is dense and m > n
+
+struct IndGraphSkinny{T <: RealOrComplex} <: IndGraph
+  m::Int
+  n::Int
+  A::AbstractArray{T,2}
+  AA::AbstractArray{T,2}
+  F::Base.LinAlg.Cholesky #LDL factorization
+end
+
+function IndGraphSkinny(A::AbstractArray{T,2}) where {T <: RealOrComplex}
+  m, n = size(A)
+  AA = A' * A
+
+  F = LinAlg.cholfact(AA + speye(n))
+  #normrows = vec(sqrt.(sum(abs2.(A), 2)))
+
+  IndGraphSkinny(m, n, A, AA, F)
+end
+
+function (f::IndGraphSkinny){T <: RealOrComplex}(x::AbstractArray{T,1}, y::AbstractArray{T, 1})
+  # the tolerance in the following line should be customizable
+  if norm(f.A * x - y, Inf) <= 1e-10
+    return 0.0
+  end
+  return +Inf
+end
+
+function prox!{T <: RealOrComplex}(
+    x::AbstractArray{T, 1},
+    y::AbstractArray{T, 1},
+    f::IndGraphSkinny,
+    c::AbstractArray{T, 1},
+    d::AbstractArray{T, 1})
+
+  x[:] = f.F \ (c + f.A' * d)
```

Comment:
I have a question regarding this line and indicator function calculation. 
Since there is `c` dimension of `n < m` I require `Array(n)` here. But in indicator function I still need `Array(m)`. If `m` is just slightly bigger than `n` - I just use one `Array(m)` for indicator and a view on first `n` of it for `prox!`. 

However, in case `m >> n` (which should not occur, but everything is possible :) ) keeping `Array(m)` will allocate unnecessary memory, since indicator is used less frequent than `prox!`.

So, I have two variants (maybe there's a better way):
1. Use `Array(m)` and write both `prox!` and indicator in an efficient way
2. Use `Array(n)` and allow allocation on each indicator call.